### PR TITLE
tests: tests for utils.convenience

### PR DESCRIPTION
**Pull request by chrisrossi**

### CODE REVIEW

File: gridded_etl_tools/utils/attributes.py [hunk 1]
```diff
@@ -284,6 +284,11 @@ def store(self, new_store):
 
         self._store = new_store
 
+    EXTREME_VALUES_BY_UNIT = {"C": (-90, 60), "K": (183.15, 333.15), "F": (-129, 140)}
```
Comment (rbanick) [inline gridded_etl_tools/utils/attributes.py:287]: We’ve changed C and F to `deg_C` and `deg_F` to be compliant with astropy, need to change this to ensure equality.

Comment (chrisrossi) [reply gridded_etl_tools/utils/attributes.py:287]: deg_K?

File: gridded_etl_tools/utils/convenience.py [hunk 2]
```diff
@@ -328,12 +326,19 @@ def get_date_range_from_file(
...
-        date_range = self.get_date_range_from_dataset(dataset)
-        dataset.close()
-        del dataset
-        gc.collect()
-        return date_range
+        try:
+            dataset = xr.open_dataset(path, backend_kwargs=backend_kwargs)
+            return self.get_date_range_from_dataset(dataset)
+
+        finally:
```
[the remark this answers is not in the document]
Comment (chrisrossi) [reply gridded_etl_tools/utils/convenience.py]: 9b5d1e4

File: gridded_etl_tools/utils/attributes.py (hunk 1 above)
Comment (rbanick) [reply gridded_etl_tools/utils/attributes.py:287]: Nope, Kelvin is not a degree, it’s an absolute unit https://sciencenotes.org/why-there-is-no-degree-in-kelvin-temperature/#:~:text=Kelvin%20doesn't%20use%20degrees,you%20include%20a%20degree%20symbol.

Comment (chrisrossi) [reply gridded_etl_tools/utils/attributes.py:287]: Well, I've been saying that wrong my entire life.

**Pull request merged**